src/core/chatRoom.js

```diff
@@ -37,6 +44,66 @@ Candy.Core.ChatRoom = function(roomJid) {
 	 * Candy.Core.ChatRoster instance
 	 */
 	this.roster = new Candy.Core.ChatRoster();
+
+	// DOM-related information for this room.
+	this.dom = {
+		id: options.domId,
+		scrollPosition: options.scrollPosition || -1,
+		tab: null,
+		form: null
+	}
+};
+
+Candy.Core.ChatRoom.prototype.setTopic = function(topic) {
+	this.room.topic = topic;
+};
+
+Candy.Core.ChatRoom.prototype.getType = function() {
+	return this.room.type;
+};
+
+Candy.Core.ChatRoom.prototype.incrementMessageCount = function() {
+	this.room.messageCount += 1;
+	return this.room.messageCount;
+};
+
+Candy.Core.ChatRoom.prototype.getUserCount = function() {
+	return Object.keys(this.roster.items).length;
+};
+
+Candy.Core.ChatRoom.prototype.getDomElement = function() {
+	return self.Room.getPane(this.room.jid);
+};
+
+Candy.Core.ChatRoom.prototype.getDomTab = function() {
+	return self.Chat.getTab(roomJid);
+};
+
+Candy.Core.ChatRoom.prototype.getDomForm = function() {
+	return self.Room.getPane(roomJid, '.message-form');
+};
+
+Candy.Core.ChatRoom.prototype.addRoomDom = function(chatRoomContainerSelector) {
+	$(chatRoomContainerSelector).append(Mustache.to_html(Candy.View.Template.Room.pane, {
+    roomId: self.dom.id,
```

Comment:
What?! My editor is setup for 2 spaces!!

/me goes off to bring hell to sublime2.